projects/kit/components/slider/slider.component.ts

```diff
@@ -1,84 +1,93 @@
-import {DOCUMENT} from '@angular/common';
 import {
     ChangeDetectionStrategy,
-    ChangeDetectorRef,
     Component,
-    forwardRef,
+    ElementRef,
+    HostBinding,
     Inject,
+    Input,
     Optional,
     Self,
 } from '@angular/core';
-import {NgControl} from '@angular/forms';
+import {NgControl, NgModel} from '@angular/forms';
+import {USER_AGENT} from '@ng-web-apis/common';
 import {
-    isNativeFocused,
-    TUI_FOCUSABLE_ITEM_ACCESSOR,
-    TuiFocusableElementAccessor,
-    TuiNativeFocusableElement,
+    CHROMIUM_EDGE_START_VERSION,
+    isEdgeOlderThan,
+    tuiDefaultProp,
 } from '@taiga-ui/cdk';
-import {AbstractTuiSlider} from '@taiga-ui/kit/abstract';
-import {TUI_FROM_TO_TEXTS} from '@taiga-ui/kit/tokens';
-import {Observable} from 'rxjs';
+import {TuiSizeS} from '@taiga-ui/core';
 
-// @dynamic
 @Component({
-    selector: 'tui-slider',
-    templateUrl: '../../abstract/slider/slider.common.template.html',
-    styleUrls: ['../../abstract/slider/slider.common.style.less', './slider.style.less'],
+    selector: 'input[tuiSlider]',
+    template: ``,
+    styleUrls: ['./slider.style.less'],
+    host: {
+        type: 'range',
+        /**
+         * For change detection.
+         * Webkit does not have built-in method for customization of filling progress (as Firefox).
+         * We draw filling of progress by `background: linear-gradient(...)` of the track.
+         * This function triggers change detection (for `fillPercentage` function) when we drag thumb of the input.
+         */
+        '(input)': '0',
+    },
     changeDetection: ChangeDetectionStrategy.OnPush,
-    providers: [
-        {
-            provide: TUI_FOCUSABLE_ITEM_ACCESSOR,
-            useExisting: forwardRef(() => TuiSliderComponent),
-        },
-    ],
 })
-export class TuiSliderComponent
-    extends AbstractTuiSlider<number>
-    implements TuiFocusableElementAccessor
-{
-    constructor(
-        @Optional()
-        @Self()
-        @Inject(NgControl)
-        control: NgControl | null,
-        @Inject(ChangeDetectorRef) changeDetectorRef: ChangeDetectorRef,
-        @Inject(DOCUMENT) documentRef: Document,
-        @Inject(TUI_FROM_TO_TEXTS) fromToTexts$: Observable<[string, string]>,
-    ) {
-        super(control, changeDetectorRef, documentRef, fromToTexts$);
+export class TuiSliderComponent {
+    @Input()
+    @HostBinding('attr.data-size')
+    @tuiDefaultProp()
+    size: TuiSizeS = 'm';
+
+    @Input()
+    @tuiDefaultProp()
+    segments = 1;
+
+    get min(): number {
+        return Number(this.elementRef.nativeElement.min);
     }
 
-    get nativeFocusableElement(): TuiNativeFocusableElement | null {
-        return this.dotRight ? this.dotRight.nativeElement : null;
+    get max(): number {
+        return Number(this.elementRef.nativeElement.max) || 100;
```

Comment:
can we move `100` to options token?

Reply:
these default values are the same as in [native \<input type="range"/>](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/Input/range#validation).

> The value won't be less than [min](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#attr-min). **The default is 0.**
The value won't be greater than [max](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#attr-max). **The default is 100.**
The value will be a multiple of [step](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#attr-step). **The default is 1.**

To be honest, this `|| 100` never be invoked because browser will set above mentioned values (if no values was set by user).

Reply:
Thus, `|| 100` is just for the case if something goes wrong)